src/bindings/rust/wrapper.cpp

```diff
@@ -1824,4 +1823,50 @@ nixl_capi_query_mem(nixl_capi_agent_t agent,
     }
 }
 
+static nixl_capi_status_t
+nixl_capi_status_from_nixl_status(nixl_status_t status) {
+    switch (status) {
+    case NIXL_SUCCESS:
+        return NIXL_CAPI_SUCCESS;
+    case NIXL_IN_PROG:
+        return NIXL_CAPI_IN_PROG;
+    case NIXL_ERR_NO_TELEMETRY:
+        return NIXL_CAPI_ERROR_INVALID_STATE;
+    default:
+        return NIXL_CAPI_ERROR_BACKEND;
+    }
+}
+
+nixl_capi_status_t
+nixl_capi_get_xfer_telemetry(nixl_capi_agent_t agent,
+                             nixl_capi_xfer_req_t req_hndl,
+                             nixl_capi_xfer_telemetry_t telemetry) {
+    if (!agent || !req_hndl || !telemetry) {
+        return NIXL_CAPI_ERROR_INVALID_PARAM;
+    }
+
+    try {
+        nixl_xfer_telem_t cpp_telemetry;
+        nixl_capi_status_t ret = nixl_capi_status_from_nixl_status(
+            agent->inner->getXferTelemetry(req_hndl->req, cpp_telemetry));
+        if (ret == NIXL_CAPI_SUCCESS) {
+            // Convert C++ telemetry structure to C structure
+            // Convert steady_clock time_point to microseconds since steady_clock epoch
+            auto steady_epoch = std::chrono::steady_clock::time_point{};
```

Comment:
Can move this logic to `agent.rs`? Like similar funcs. Then can remove struct and `nixl_capi_status_from_nixl_status`